File: crates/flowrt-codegen/tests/golden/bounded_channel_iox2_cpp/expected/app/stubs/cpp/source.cpp
// FlowRT 管理参考模板（C++）。可删除重建；复制到用户 app/ 后再修改。

#include "flowrt_app/runtime_shell.hpp"

#include <memory>

namespace {

class Source final : public flowrt_app::SourceInterface {
public:
    flowrt::Status on_tick(
        flowrt::Output<flowrt_app::Packet>& packet) override {
        packet.write(flowrt_app::Packet{});
        return flowrt::ok();
    }
};

}  // namespace

namespace flowrt_user {

flowrt_app::App build_app() {
    return flowrt_app::App(std::make_unique<Source>());
}

}  // namespace flowrt_user

File: crates/flowrt-codegen/tests/golden/bounded_channel_iox2_rust/expected/app/stubs/rust/source.rs
// FlowRT 管理参考模板（Rust）。可删除重建；复制到用户 app/ 后再修改。

#[derive(Default)]
pub struct Source;

impl flowrt_app::components::Source for Source {
    fn on_tick(
        &mut self,
        packet: &mut flowrt::Output<flowrt_app::messages::Packet>,
    ) -> flowrt::Status {
        packet.write(flowrt_app::messages::Packet::default());
        flowrt::Status::Ok
    }
}

pub fn build_app() -> flowrt_app::runtime_shell::App {
    flowrt_app::runtime_shell::App::new(Box::new(Source::default()))
}
